src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchOptions.cs

```diff
@@ -49,41 +49,77 @@ internal sealed class FindReferencesSearchOptions
             Cascade = cascade;
             Explicit = @explicit;
+            UnidirectionalHierarchyCascade = unidirectionalHierarchyCascade;
         }
 
         public FindReferencesSearchOptions With(
             Optional<bool> associatePropertyReferencesWithSpecificAccessor = default,
             Optional<bool> cascade = default,
-            Optional<bool> @explicit = default)
+            Optional<bool> @explicit = default,
+            Optional<bool> unidirectionalHierarchyCascade = default)
         {
             var newAssociatePropertyReferencesWithSpecificAccessor = associatePropertyReferencesWithSpecificAccessor.HasValue ? associatePropertyReferencesWithSpecificAccessor.Value : AssociatePropertyReferencesWithSpecificAccessor;
             var newCascade = cascade.HasValue ? cascade.Value : Cascade;
             var newExplicit = @explicit.HasValue ? @explicit.Value : Explicit;
+            var newUnidirectionalHierarchyCascade = unidirectionalHierarchyCascade.HasValue ? unidirectionalHierarchyCascade.Value : UnidirectionalHierarchyCascade;
 
             if (newAssociatePropertyReferencesWithSpecificAccessor == AssociatePropertyReferencesWithSpecificAccessor &&
                 newCascade == Cascade &&
-                newExplicit == Explicit)
+                newExplicit == Explicit &&
+                newUnidirectionalHierarchyCascade == UnidirectionalHierarchyCascade)
             {
                 return this;
             }
 
-            return new FindReferencesSearchOptions(newAssociatePropertyReferencesWithSpecificAccessor, newCascade, newExplicit);
+            return new FindReferencesSearchOptions(
+                newAssociatePropertyReferencesWithSpecificAccessor,
+                newCascade,
+                newExplicit,
+                newUnidirectionalHierarchyCascade);
         }
 
         /// <summary>
-        /// For IDE features, if the user starts searching on an accessor, then we want to give
-        /// results associated with the specific accessor.  Otherwise, if they search on a property,
-        /// then associate everything with the property.
+        /// For IDE features, if the user starts searching on an accessor, then we want to give results associated with
+        /// the specific accessor.  Otherwise, if they search on a property, then associate everything with the
+        /// property.  We also only want to travel an inheritance hierarchy unidirectionally so that we only see potential
+        /// references that could actually reach this particular member.
         /// </summary>
         public static FindReferencesSearchOptions GetFeatureOptionsForStartingSymbol(ISymbol symbol)
-            => Default.With(associatePropertyReferencesWithSpecificAccessor: symbol.IsPropertyAccessor());
+            => Default.With(associatePropertyReferencesWithSpecificAccessor: symbol.IsPropertyAccessor(),
+                            unidirectionalHierarchyCascade: true);
```

Comment:
❔ Is this method not used for rename scenarios? If not, consider updating the comment:

> ... that could actually reach this particular member **(this method is not used for rename scenarios)**.

Reply:
will doc.  it is not.  `GetFeatureOptionsForStartingSymbol` means: Get teh options for the 'find refs FEATURE'.  The name is not good.  Would def appreciate any thoughts on that.